kong/plugins/api-response-merger/body_transformer.lua

```diff
@@ -157,50 +143,44 @@ local function set_in_table(path, upstream_body, value, resource)
         -- just skip
       else
         -- print('key_to_dest_resource: ' .. key_to_dest_resource .. ', _dest_resource:' .. dump(_dest_resource))
-        if type(key_to_dest_resource) == 'number' then key_to_dest_resource = key_to_dest_resource + 1 end
-        if j == #dest_resource_path then _parent_of_dest_resource = _dest_resource end
+        if type(key_to_dest_resource) == 'number' then
+          key_to_dest_resource = key_to_dest_resource + 1
+        end
+        if j == #dest_resource_path then
+          _parent_of_dest_resource = _dest_resource
+        end
         _dest_resource = _dest_resource[key_to_dest_resource]
         _external_key_to_dest_resource = key_to_dest_resource
       end
     end
 
-    -- result is an array and we want to get single ID to response
-    --- XXX: resource can be nil
-    if resource ~= nil and resource.config.search_in_array and utils.is_array(value) then
```

Comment:
As you are removing the usage of `search_in_array` please remove it also from config schema.

Reply:
I cannot, since this is not backward compatible change. However, I need to sit with you and consult what functionality can be changed or rewritten here.

Reply:
You can, while updating the submodule in helm-kong you can remove occurrence of `search_in_array`